packages/web-components/fast-components/src/slider-label/slider-label.ts

```diff
@@ -25,63 +26,86 @@ export class SliderLabel extends FASTElement {
 
     @attr({ attribute: "hide-mark", mode: "boolean" })
     public hideMark: boolean = false;
-    public config: SliderConfiguration = {
-        min: 0,
-        max: 0,
-        direction: Direction.ltr,
-        orientation: SliderOrientation.horizontal,
-    };
 
-    @attr({ mode: "boolean" })
-    public disabled: boolean;
+    @attr({ attribute: "disabled", mode: "boolean" })
+    public disabled: boolean; // Map to proxy element
+
+    @observable
+    public sliderOrientation: Orientation;
+    @observable
+    public sliderMinPosition: number;
+    @observable
+    public sliderMaxPosition: number;
+    @observable
+    public sliderDirection: Direction = Direction.ltr;
 
     public connectedCallback(): void {
         super.connectedCallback();
         this.getSliderConfiguration();
-        this.setStyleForOrientation();
         this.positionStyle = this.positionAsStyle();
-    }
+        const notifier = Observable.getNotifier(this.parentNode as FASTSlider);
+        const handler = {
+            sliderLabel: this,
+            handleChange(source: any, propertyName: string) {
+                switch (propertyName) {
+                    case "direction":
+                        this.sliderLabel.direction = source.direction;
+                        break;
+                    case "orientation":
+                        this.sliderLabel.sliderOrientation = source.orientation;
+                        break;
+                    case "max":
+                        this.sliderLabel.max = source.max;
+                        break;
+                    case "min":
+                        this.sliderLabel.min = source.min;
+                        break;
+                    default:
+                        break;
+                }
+                this.sliderLabel.positionStyle = this.sliderLabel.positionAsStyle();
+            },
+        };
```

Comment:
I think this *could* cause a memory leak because you're not maintaining the object reference to unsubscribe when the component disconnects.

To circumvent you could either:
1. store this handler object on the element instance
2. give this element a `handleChange` method and subscribe the instance itself: `notifier.subscribe(this, "min")`

Then unsubscribe on tear-down

Reply:
Unfortunately, no weak ref magic. I wish, but it's not in the language yet. @nicholasrice is correct in that these handlers will need to be torn down. You can just store the notifier and the handler and then unsubscribe in the disconnectedCallback.

Reply:
I think my recent push does all the recommended things, and bonus, it still works :)